optax/_src/loss.py

```diff
@@ -503,3 +503,19 @@ def kl_divergence_with_log_targets(log_predictions: chex.Array,
   chex.assert_type([log_predictions, log_targets], float)
   loss = jnp.exp(log_targets) * (log_targets - log_predictions)
   return jnp.sum(loss, axis=-1)
+
+def hinge_loss(predictions: chex.Array, targets: chex.Array) -> chex.Array:
```

Comment:
Please add tests for this function that compare against analytically calculated values, ideally for several different array ranks and shapes. If you wrap the test in `chex.all_variants` you can test all the jax-transformed versions of the function (see the other tests for how to do this). Thanks!